packages/wrangler/src/deployment-bundle/node-compat.ts

```diff
@@ -118,7 +118,7 @@ export function getNodeCompatMode({
 	}
 
 	return {
-		legacy: node_compat === true,
+		legacy: mode === "legacy",
```

Comment:
why this change?

Reply:
Sorry, should have mentioned. This is just a really minor nit that I had which was because it was duplicating some logic from above, whereas this change derives the property from the above if/else.

Reply:
I've added some tests here because I believe that was actually fixing a bug: https://github.com/cloudflare/workers-sdk/pull/6554/commits/9d8e1a72a2ea74c43b75441583011d08303a0e83

Unless I'm mistaken and it should return `legacy` no matter what flags are overriding the `node_compat = true` setting.

Reply:
I have some refactoring of this code in here: https://github.com/cloudflare/workers-sdk/pull/6545

Reply:
Ah I see. Coincidence there. Would it be best if I just removed all the refactoring I've done here and leave the original change (just the bug fix)?

Reply:
Up to you. That other PR won't land for a while.
So no doubt there will be other conflicts to tidy up then.

Reply:
I'll just remove it - they both tackle the same issue.